zarr/core.py

```diff
@@ -424,6 +424,36 @@ def __array__(self, *args):
         return a
 
     def islice(self, start=None, end=None):
+        """
+        Yield a generator for iterating over the entire or parts of the
+        array.
```

Comment:
It's probably worthwhile saying why this is useful and more efficient than iterating directly over a slice.